STIPS/STIP-007.md

```diff
@@ -0,0 +1,145 @@
+| TradeModule    	| function getTradeCalldata( address _sourceToken, address _destinationToken, address _destinationAddress, uint256 _sourceQuantity, uint256 _minDestinationQuantity, bytes memory _data) 	| `_data` parameter encodes special info depending on the integration, for 0x it encodes the whole trade, for UniswapV3 it encodes the fee 	|
+| WrapModule     	| function getWrapCallData( address _underlyingToken, address _wrappedToken, uint256 _underlyingUnits, address _to, bytes memory _wrapData)                                              	| `_wrapData` param has not been used in the wild but could be used to encode some arbitrary data.                                         	|
+| AmmModule      	| function getProvideLiquidityCalldata( address _setToken, address _pool, address[] calldata _components, uint256[] calldata _maxTokensIn, uint256 _minLiquidity)                        	| `setToken` parameter is only used for a balance check and then as a `to` address, which would be compatible with this implementation.    	|
+
+You will notice that these different adapters have the same general pattern in common:
+1. There is a source token(s) and an amount of source token to transform (we can make the source tokens/amounts param an array to support `AmmModule`)
+2. There is a destination token and an expected minimum amount to get back (except WrapModule where we could 0 it out)
+3. There is a recipient address (`TradeModule`: `destinationAddress`, `WrapModule`: `_to`, `AmmModule`: `_setToken`)
+
+## Open Questions
+- [ ] Do we set a defined order of actions? (ie trades, wraps, LPs)
+- [ ] Do we utilize our adapters to facilitate the transformations?
+- [ ] What level of decentralization is necessary? Can we just pass in parsable bytedata or do trades have to be figured out all on-chain?
+    - *Start with decentralization as a lower priority. Ok with relying on centralized API to generate the necessary transformations*
+## Feasibility Analysis
+Provide potential solution(s) including the pros and cons of those solutions and who are the different stakeholders in each solution. A recommended solution should be chosen here.
+### Option 1: Use current adapters to create invokable bytedata
+**Summary:** Pass in standardized inputs which point to an adapter that generates bytecode that is invoked by the wallet
+
+**Pros:**
+- Pushes most of the work off-chain to figuring out how many assets to trade for, what to wrap, etc.
+- Allows us to be sure that we're doing transformations that are supported by the protocol
+- Fairly quick to market
+- Inputs are far more human-readable
+
+**Cons:**
+- Reliant on a more centralized off-chain API
+- May be issues with on-going compatibility since adapters are built more for modules
+- May require additional contracts/code to turn standardized data inputs into a function call to the adapter
+
+### Option 2: Use 0x-like implementation
```

Comment:
One potential problem with this implementation is when a set contains wrapped tokens. At the time of submitting a transaction, a set may contain 1.001 DAI worth of cDAI. However, by the time the transaction is mined, the exchange rate will have likely increased, making minting the set require 1.002 DAI worth of cDAI.

Since all trades and transformations would be computed off-chain, this solution would only purchase 1.001 DAI, and would then revert when attempting to wrap it into the correct about of cDAI.